src/averageAges.js

```diff
@@ -55,7 +69,23 @@ function calculateWomenAverageAge(people, withChildren) {
  * @return {number}
  */
 function calculateAverageAgeDiff(people, onlyWithSon) {
-  // write code here
+  const women = people.filter(woman => (
+    woman.sex === 'f'
+    && people.some(child => child.mother === woman.name))
+  );
+
+  const childrens = people.filter(child => (
+    onlyWithSon
+      ? child.sex === 'm' && people.some(mother => child.mother === mother.name)
+      : people.some(mother => child.mother === mother.name)
+  ));
+
+  const age = childrens.map(child => (
+    child.born - women.find(mother => mother.name === child.mother).born)
```

Comment:
you can use people and delete getting women (-1 cycle)
```suggestion
    child.born - people.find(mother => mother.name === child.mother).born)
```